tests/modules/sightings/test_models.py

```diff
@@ -1,48 +1,101 @@
 # -*- coding: utf-8 -*-
 # pylint: disable=invalid-name,missing-docstring
-import json
+import json, logging
 
 
-def test_sighting_create_and_destroy(flask_app_client, regular_user):
+def test_sighting_create_and_add_encounters(db):
 
     from app.modules.sightings.models import Sighting
+    from app.modules.encounters.models import Encounter
 
-    with flask_app_client.login(regular_user, auth_scopes=('sightings:write',)):
-        response = flask_app_client.post(
-            '/api/v1/sightings/',
-            data=json.dumps(
-                {
-                    'title': 'Testing generated.',
-                }
-            ),
-        )
-
-    assert response.status_code == 200
-    assert response.content_type == 'application/json'
-    assert isinstance(response.json, dict)
-    assert set(response.json.keys()) >= {
-        'guid',
-        'title',
-    }
-
-    sighting_guid = response.json['guid']
-    read_sighting = Sighting.query.get(response.json['guid'])
-    assert read_sighting.title == 'Testing generated.'
-
-    with flask_app_client.login(regular_user, auth_scopes=('sightings:read',)):
-        response = flask_app_client.get('/api/v1/sightings/%s' % sighting_guid)
-
-    assert response.status_code == 200
-    assert response.content_type == 'application/json'
-    assert isinstance(response.json, dict)
-    assert set(response.json.keys()) >= {
-        'guid',
-        'title',
-    }
-
-    with flask_app_client.login(regular_user, auth_scopes=('sightings:write',)):
-        response = flask_app_client.delete('/api/v1/sightings/%s' % sighting_guid)
-
-    assert response.status_code == 204
-    read_sighting = Sighting.query.get(sighting_guid)
-    assert read_sighting is None
+    test_sighting = Sighting(
+        title='New Test Sighting',
+    )
+
+    test_encounter_a = Encounter(
+        title='New Test Encounter A',
+    )
+
+    test_encounter_b = Encounter(
+        title='New Test Encounter B',
+    )
+
+    with db.session.begin():
+        db.session.add(test_encounter_a)
+        db.session.add(test_sighting)
+
+    db.session.refresh(test_encounter_a)
+    db.session.refresh(test_sighting)
+
+    assert len(test_sighting.get_encounters()) == 0
+    assert test_encounter_a.get_sighting() is None
+
+    test_sighting.add_encounter(test_encounter_a)
+
+    with db.session.begin():
```

Comment:
test_encounter_a and test_sighting were added to the db above, they shouldn't need readiing once linked together.... I don't think.

Reply:
Cool. Will get some clarification on db transaction details in our next meeting, this was kinda hacked in and left in because it worked.